core/market/src/negotiation/common.rs

```diff
@@ -333,6 +336,23 @@ impl CommonBroker {
             &agreement.id,
             reason.display(),
         );
+
+        // If we are Requestor we would like to be able to negotiate with the same
+        // provider for the second time, so we must generate new Proposal for him.
+        // Note: Regeneration failure isn't propagated to Requestor, because termination
+        // succeeded at this point.
+        if let OwnerType::Requestor = agreement.id.owner() {
+            self.regenerate_proposal(&agreement)
```

Comment:
it should be spawned to the side task, not to hold terminate agreement op, which should be as fast as possible

whats more, whole this logic should be enclosed within `regenerate_proposal`

Reply:
I see you've addressed it